eldritch/test_combat.py

```diff
@@ -1603,6 +1603,81 @@ def testCannotUseEnchantWeaponInNextCombat(self):
       self.assertEqual(rand.call_count, 2)  # Fight (4) + spawn (-2) + physical immunity (0)
 
 
+class CombatWithMagicPowderTest(EventTest):
+  def setUp(self):
+    super().setUp()
+    self.powder = items.MagicPowder(0)
+    self.char.possessions.append(self.powder)
+
+  def start(self, monster):
+    # pylint: disable=attribute-defined-outside-init
+    self.combat = Combat(self.char, monster)
+    self.state.event_stack.append(self.combat)
+
+    fight_or_flee = self.resolve_to_choice(MultipleChoice)
+    fight_or_flee.resolve(self.state, "Fight")
+
+    return self.resolve_to_choice(CombatChoice)
+
+  def testUsePowder(self):
+    self.assertEqual(self.char.sanity, 5)
+    cultist = monsters.Cultist()
+    combat_choice = self.start(cultist)
+    combat_choice.resolve(self.state, "Magic Powder0")
+    combat_choice.resolve(self.state, "done")
+    # Mock shows that we get enough dice to actually need the powder
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(side_effect=[1]*8+[5]*100)):
```

Comment:
Alternate suggestion: `new=mock.MagicMock(return_value=5) as rand`
followed by
```
self.assertEqual(rand.call_count, expected_dice)
```
expected dice is probably 14 (fight of 4 + 1 from cultist + 9 from powder)